lib/private/Security/IdentityProof/Manager.php

```diff
@@ -68,7 +69,16 @@ protected function generateKeyPair(): array {
 
 		// Generate new key
 		$res = openssl_pkey_new($config);
-		openssl_pkey_export($res, $privateKey);
+
+		if ($res === null) {
+			$this->logOpensslError();
+			throw new \RuntimeException('OpenSSL reported a problem');
+		}
+
+		if (openssl_pkey_export($res, $privateKey, null, $config) === false) {
+			$this->logOpensslError();
+			throw new \RuntimeException('OpenSSL reported a problem');
+		}
 
 		// Extract the public key from $res to $pubKey
 		$publicKey = openssl_pkey_get_details($res);
```

Comment:
```suggestion
		$publicKey = openssl_pkey_get_details($res);
		if ($publicKey === false) {
			$this->logOpensslError();
			throw new \RuntimeException('OpenSSL reported a problem');
		}
```

Not sure if we need this. We don't have it at `PublicKeyTokenProvider`.

Reply:
Lets also do this if we move it to some generic method later.